### PR TITLE
Use roxygen markdown for help documentation

### PR DESCRIPTION
**Pull request by andrie**
This changes the help documentation to use roxygen markdown, since this is in general easier to write and maintain.

In addition, moves documentation examples into inst/examples/___.R - this makes it easier to edit documentation examples, since you can run the R code directly.

**DoD**

- [ ] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [x] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [x] No new error or warning messages are introduced.

- [x] All interaction with a semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [x] If the change affects code or repo sctructure, README, documentation and code comments should be updated. 

- [ ] All code has been peer-reviewed before merging into any main branch.

- [ ] All changes have been merged into the main branch we use for development (develop).

- [ ] Continuous integration checks (linter, unit tests) are configured and passed.

- [ ] Unit tests added for all new or changed logic.

- [ ] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [ ] Any added or touched code follows our style-guide.

### CODE REVIEW

File: man/search_selection_choices.Rd [hunk 1]
```diff
@@ -15,7 +15,7 @@ search_selection_choices(
 )
 }
 \arguments{
-\item{input_id}{Input name. Reactive value is available under input[[input_id]].}
+\item{input_id}{Input name. Reactive value is available under input[\link{input_id}].}
```
Comment (dokato) [inline man/search_selection_choices.Rd:18]: Why this changed to `\link`

**Review by dokato — COMMENTED** [review]
Thanks @andrie . This really improves the readability of the code. One thing that is puzzling is why GitHub action didn't run for this PR, but that's for me to figure out.

Another thing is, that all square brackets `[]` became a link in the `md`. Not sure if this is related to `Roxygen: list(markdown = TRUE)` option? But now I get in R checks many warnings:

```
I have a questions about Missing link or links in documentation object 'dropdown_input.Rd':
    ‘input_id’
```
Probably we should change these places  to `\code{input[[input_id]]}`

File: man/search_selection_choices.Rd (hunk 1 above)
Comment (andrie) [reply man/search_selection_choices.Rd:18]: All elements inside `[...]` will change to links in the Roxygen Markdown format.  This one should not have happened, and the proper way to write roxygen markdown is to use code blocks, i.e. backticks for code blocks.  I'm not sure if this is going to appear correctly in github comments, but ``input[input_id]]``

Comment (andrie) [reply man/search_selection_choices.Rd:18]: The backticks disappeared, so substitute quotes for backticks in this workaround `'input[input_id]]'`

Comment (dokato) [reply man/search_selection_choices.Rd:18]: I think this would do `\code{input[[input_id]]}`, wouldn't it?

**Pull request closed without merging**